### DISCUSSION TITLE
Loading Roberta Model

### DISCUSSION THREAD

**Discussion by Tacacs-1101** [deepjavalibrary/djl#1175] [Q&A]
Hi, I want to load Roberta model in a java environment. The trained model can be loaded in DJL but how to tokenize the input sentence during inference in DJL as Roberta uses a different tokenizer than BERT.

**Comment by zachgk** [comment]
If you want your model to work right, you need to give it input data with the exact same format as the input data it was trained on. So, the best way is if you implement the Roberta tokenizer in Java. You can look at our [nlp folder](https://github.com/deepjavalibrary/djl/tree/master/api/src/main/java/ai/djl/modality/nlp) to see some of the utilities we have for working with preprocessing and tokenizers, especially the [nlp pre-processing folder](https://github.com/deepjavalibrary/djl/tree/master/api/src/main/java/ai/djl/modality/nlp/preprocess).

Alternatively, you could see if something like [sentencepiece](http://docs.djl.ai/extensions/sentencepiece/index.html) will work or find some way to call python from Java like Jython.
